shared-spring/src/main/java/tds/shared/spring/interceptors/RestTemplateLoggingInterceptor.java

```diff
@@ -34,18 +35,25 @@ public RestTemplateLoggingInterceptor(final ObjectMapper objectMapper) {
     public ClientHttpResponse intercept(final HttpRequest request,
                                         final byte[] body,
                                         final ClientHttpRequestExecution clientHttpRequestExecution) throws IOException {
-        final UUID traceId = UUID.randomUUID();
+
+        // get tracer id that used to associate all rest calls that are a child of this request
+        final String traceId = Optional.fromNullable(MDC.get(TRACER_ID_HEADER)).or(UUID.randomUUID().toString());
         logRequest(request, body, traceId);
-        ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
-        logResponse(response, traceId);
-        return response;
+        try {
+            ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
+            logResponse(response, traceId);
+            return response;
+        } catch (IOException e) {
+            log.error(String.format("Exception occurred while executing rest request. METHOD: %s URI: %s", request.getMethod(), request.getURI()), e);
+            throw e;
```

Comment:
So it looks like `logResponse` can throw the `IOException`.  So that means the catch is for when the logging has an error not when the execution of the httpRequest throws an error.  So if the logging method throws an error and it bubbles up, then it would be changing the flow of the application.

Should the try/catch be just around the `logResponse` call and then it should swallow it, therefore if the `execute()` call throws and error it bubbles up as it should.  If the current try/catch is changed to not throw, then if the client http request execute throws an `IOException` it won't bubble and it should in that case, but not in the case when the `logResponse` throws an `IOException`.